mobile-widgets/qmlmanager.cpp

```diff
@@ -1056,15 +1056,30 @@ void QMLManager::commitChanges(QString diveId, QString date, QString location, Q
 	}
 	// gasmix for first cylinder
 	if (myDive->firstGas() != gasmix) {
-		int o2 = parseGasMixO2(gasmix);
-		int he = parseGasMixHE(gasmix);
-		// the QML code SHOULD only accept valid gas mixes, but just to make sure
-		if (o2 >= 0 && o2 <= 1000 &&
-		    he >= 0 && he <= 1000 &&
-		    o2 + he <= 1000) {
-			diveChanged = true;
-			d->cylinder[0].gasmix.o2.permille = o2;
-			d->cylinder[0].gasmix.he.permille = he;
+		QStringList gaslist;
+		if (gasmix.contains(",")){
+			gaslist = gasmix.split(QRegExp("\\s*,\\s*"));
```

Comment:
Current qml code has a validator restricting the input to two digits or EANXX, etc. 

With that you cannot enter multiple gas mixes using the UI.

Reply:
Yeah, I noticed that. If you have a DC that has reported the usage of multiple gases you can edit them, as long as you don't remove the commas.